src/wasm-lib/kcl/src/execution/mod.rs

```diff
@@ -2279,6 +2279,51 @@ fn assign_args_to_params(
     Ok(fn_memory)
 }
 
+fn assign_args_to_params_kw(
+    function_expression: NodeRef<'_, FunctionExpression>,
+    mut args: crate::std::args::KwArgs,
+    mut fn_memory: ProgramMemory,
+) -> Result<ProgramMemory, KclError> {
+    // Add the arguments to the memory.  A new call frame should have already
+    // been created.
+    let source_ranges = vec![function_expression.into()];
+    for param in function_expression.params.iter() {
+        if param.labeled {
+            let arg = args.labeled.get(&param.identifier.name);
+            let arg_val = match arg {
+                Some(arg) => arg.value.clone(),
+                None => match param.default_value {
+                    Some(ref default_val) => KclValue::from(default_val.clone()),
+                    None => {
+                        return Err(KclError::Semantic(KclErrorDetails {
+                            source_ranges,
+                            message: format!(
+                                "This function declares a parameter {}, but you haven't passed it one.",
+                                param.identifier.name
+                            ),
+                        }));
+                    }
+                },
+            };
+            fn_memory.add(&param.identifier.name, arg_val, (&param.identifier).into())?;
+        } else {
+            let Some(unlabeled) = args.unlabeled.take() else {
```

Comment:
I'm more just thinking out loud. It's not about this PR. I wish unlabeled weren't so special. We need to do things twice all over to handle unlabeled. Could there be a special label that has no syntax? Parsing would still need to treat it specially, but then maybe it could be treated identically to labeled args in execution?

Reply:
I guess we could change the parameter list to be `HashMap<Option<String>, Arg>` instead of `(Option<Arg>, HashMap<String, Arg>)`. What I like about the current design is that it compile-time enforces that there's 0 or 1 unlabeled args, but my alternative design allows any number -- and then you have to check the number at runtime.

But it would be elegant. Idk. Which do you prefer?

Reply:
Don't worry about it. Keep it as is. I was trying to see if we could avoid duplicating code by abstracting over it when they're almost identical.